### DISCUSSION TITLE
Build position from Cartesian coordinates

### DISCUSSION THREAD

**Discussion by facorazza** [skyfielders/python-skyfield#702] [Q&A] [answered]
I'm looking for a way to find the elevation the azimuth and the distance of a moving object w.r.t a fixed station on the surface of the Earth.
Ideally, I'd like to do something like:
```
elevation, azimuth, distance = (moving_object_position - ground_station_position).at(t).altaz()
```
with the position of the ground station defined as:
```
ground_station_position = wgs84.latlon(50, 50, elevation_m=50)
```
However, I'm not sure how to build the position for the moving object. The data I have is a Cartesian position in meters computed from the center of the Earth in a fixed frame of reference (as better explained [here](https://help.agi.com/stk/11.0.1/Content/stk/AstroComps.htm#CentralBodyCoordinateSystems)).
In Python, it looks like this: `[[datetime, x, y, z, vx, vy, vz], ..., [...]]`
At first, I tried to initialize the position using the Geocentric class which returns a GeographicPosition object.
```
moving_object_position = Geocentric([x / AU_M, y / AU_M, z / AU_M], t=t)
```
However, I got a bit confused by the fact that this object has no position attribute and I cannot subtract the ground station position. Then I added the `frame_xyz(itrs)` which should also be the correct frame of reference for my data, but it returns a Distance object that cannot be subtracted from the GeographicPosition of the ground_station.
Finally, I tried to initialize an ITRS position for each datetime data point using the ITRSPosition class:
```
moving_object_position = ITRSPosition(Distance(m=[x, y, z]))
```
This method seems to work, but the resulting elevation, azimuth and distance are not as expected. At this point, I'm a bit confused by all the frames of reference available, so it could also be that I'm using the wrong class for the job. If anyone can point me in the right direct I'd appreciate it.

**Comment by brandon-rhodes** [comment]
> This method seems to work, but the resulting elevation, azimuth and distance are not as expected.

Given how you describe your coordinates, it sounds like `ITRSPosition` is the class you would want to use for one of your Fixed positions. But maybe the STK library uses a different coordinate system from the Earth than the ITRS? I cannot tell from its documentation (thank you for linking to it, though). “Earth realizes its Fixed frame from the transformation algorithm between it and the ICRF” — while that's true, it doesn't specify which x,y,z system it's using for coordinates that are fixed to the Earth's surface.

Feel free to share a small script that creates a `wgs84.latlon()` observer, then an `ITRSPosition` target, and computes the offset between them. Explain what offset you would like to receive instead, and we can try to figure out why that's not what you are getting.

**Comment by facorazza** [comment, marked as answer]
The ITRSPosition is indeed the correct class. After plotting the altitude, the azimuth and the distance I realized what was wrong. For the ground station position, I mixed East and West and forgot to put a minus sign in the longitude.

**Reply by brandon-rhodes** [reply]
Thanks for reporting back about your results. I'm glad you were able to get it working!
